sql/snowflake/SnowflakeLexer.g4

```diff
@@ -1036,6 +1037,7 @@ VOLATILE:                                              'VOLATILE';
 WAREHOUSE:                                             'WAREHOUSE';
 WAREHOUSES:                                            'WAREHOUSES';
 WAREHOUSE_SIZE:                                        'WAREHOUSE_SIZE';
+WAREHOUSE_TYPE:                                        'WAREHOUSE_TYPE';
```

Comment:
I guess it should be added to [non_reserved_words](https://github.com/antlr/grammars-v4/blob/master/sql/snowflake/SnowflakeParser.g4#L3557C1-L3557C1) since it could be an identifier. The same for `SNOWPARK_OPTIMIZED`.

Reply:
`SNOWPARK_OPTIMIZED` contains single quotes and a dash